src/Environment.hpp

```diff
@@ -40,8 +42,9 @@ namespace geopm
             virtual std::string profile(void) const = 0;
             virtual std::string frequency_map(void) const = 0;
             virtual std::string agent(void) const = 0;
-            virtual std::string trace_signals(void) const = 0;
-            virtual std::string report_signals(void) const = 0;
+            virtual std::vector<std::pair<std::string, int> > trace_signals(void) const = 0;
+            virtual std::vector<std::pair<std::string, int> > report_signals(void) const = 0;
+            virtual std::vector<std::pair<std::string, int> > signal_parser(std::string environment_variable_contents) const = 0;
```

Comment:
signal_parser belongs in EnvironmentImp, not in the public interface of Environment.